core/src/client/mod.rs

```diff
@@ -734,4 +740,72 @@ impl StarknetClient for StarknetClientImpl {
 
         Ok(Address::from(evm_address_sliced))
     }
+
+    /// Returns token balances for a specific address given a list of contracts.
+    ///
+    /// # Arguments
+    ///
+    /// * `address(Address)` - specific address
+    /// * `contract_addresses(Vec<Address>)` - List of contract addresses
+    ///
+    /// # Returns
+    ///
+    ///  * `token_balances(TokenBalances)` - Token balances
+    ///
+    /// `Ok(<TokenBalances>)` if the operation was successful.
+    /// `Err(KakarotClientError)` if the operation failed.
+    async fn get_token_balances(
+        &self,
+        address: Address,
+        contract_addresses: Vec<Address>,
+    ) -> Result<TokenBalances, KakarotClientError> {
+        let mut token_balances = Vec::new();
+        let starknet_user_address = self
+            .compute_starknet_address(address, StarknetBlockId::Tag(BlockTag::Latest))
+            .await?;
+        let entrypoint = FieldElement::from_str("balanceOf").map_err(|e| {
```

Comment:
the entrypoint should be encoded as an EVM entrypoint for kakarot execute_at_address 

`keccak256("balanceOf(address)")`

https://solidity-by-example.org/function-selector/